state/execution.go

```diff
@@ -239,6 +249,19 @@ func (blockExec *BlockExecutor) Commit(
 		"app_hash", fmt.Sprintf("%X", res.Data),
 	)
 
+	if blockExec.sidecar != nil {
+		// update sidecar mempool
+		err = blockExec.sidecar.Update(
```

Comment:
The mempool is locked before updated, why the sidecar isn't?

Reply:
UPDATE: Agreed to take a closer look and try to find concerning scenarios.

Reply:
Fixed in https://github.com/skip-mev/mev-tendermint/pull/126